vm-virtio/src/rng.rs

```diff
@@ -296,16 +325,14 @@ impl VirtioDevice for Rng {
                 interrupt_cb,
                 queue_evt: queue_evts.remove(0),
                 kill_evt,
+                pause_evt,
             };
 
-            let worker_result = thread::Builder::new()
+            thread::Builder::new()
                 .name("virtio_rng".to_string())
-                .spawn(move || handler.run());
-
-            if let Err(e) = worker_result {
-                error!("failed to spawn virtio_rng worker: {}", e);
-                return Err(ActivateError::BadActivate);
-            }
+                .spawn(move || handler.run())
+                .map(|thread| self.epoll_thread = Some(thread))
```

Comment:
Is the thread going to return when you park it? I'm asking because I thought we could simply use `mpsc` or `AtomicBool` to notify it that it needs to resume. I think that's the approach you already used with vcpu threads.

Reply:
With vcpu threads, we use an atomic boolean to handle both the vcpu thread exit from `run()` and the potential spurious returns from `park()`. We need to toggle that boolean from the `pause/resume` methods. Will fix.